frontend/src/v5/ui/components/dashboard/dashboardList/dashboardListItem/shareModal/shareModal.styles.tsx

```diff
@@ -0,0 +1,138 @@
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import styled from 'styled-components';
+import { TextField, withStyles } from '@material-ui/core';
+import Tooltip, { TooltipProps } from '@material-ui/core/Tooltip';
+import copyToClipboardIcon from '@assets/icons/copy_to_clipboard';
+import tick from '@assets/icons/tick';
+
+export const Container = styled.div`
+	display: flex;
+	flex-grow: 1;
+	flex-direction: column;
+	justify-content: center;
+`;
+
+export const LinkLabel = styled.span`
+	${({ theme }) => theme.typography.kicker};
+	color: ${({ theme }) => theme.palette.base.main};
+	font-family: ${({ theme }) => theme.typography.fontFamily};
+`;
+
+export const LinkContainer = styled.div`
+	display: flex;
+	flex-direction: row;
+	justify-content: space-between;
+	width: 100%;
+	max-width: 100%;
+	padding: 10px 15px;
+	border-radius: 5px;
+	border: 1px solid ${({ theme }) => theme.palette.base.lightest};
+	box-sizing: border-box;
+	
+	:hover {
+		cursor: pointer;
+	}
+`;
+export const Link = styled.span`
+	${({ theme }) => theme.typography.body1};
+	color: ${({ theme }) => theme.palette.base.main};
+	font-family: ${({ theme }) => theme.typography.fontFamily};
+	display: flex;
+	align-items: center;
+`;
+
+export const MailToButton = styled.a`
+	${({ theme }) => theme.typography.link};
+	color: ${({ theme }) => theme.palette.base.main};
+	font-family: ${({ theme }) => theme.typography.fontFamily};
+	margin-top: 20px;
+	width: fit-content;
+`;
+
+export const CopyToClipboardIcon = styled(copyToClipboardIcon)``;
+
+export const Tick = styled(tick)``;
+
+const SVG_PADDING_IN_PX = 9;
+
+export const CopyToClipboardTooltip = withStyles(() => ({
+	popper: {
+		marginTop: -SVG_PADDING_IN_PX,
+	},
+}))(Tooltip);
+
+type CopiedToClipboardTooltipProps = TooltipProps & {
+	theme: any
+};
+
+export const CopiedToClipboardTooltip = styled(
+	({ className, theme, ...props }: CopiedToClipboardTooltipProps) => (
+		<CopyToClipboardTooltip
+			{...props}
+			PopperProps={{ keepMounted: true }}
+			classes={{ popper: className }}
+		/>
+	),
+)`
+	& .MuiTooltip-tooltip {
+		background-color: ${({ theme }) => theme.palette.primary.main};
+	}
+`;
+
+export const CopyToClipboardIconContainer = styled.div`
+	display: grid;
+	padding: ${SVG_PADDING_IN_PX}px;
+`;
+
+export const UrlContainer = styled(TextField)`
+	margin: 6px 0 0 0;
+
+	.MuiInputBase-root {
+		cursor: pointer;
+		margin-top: 0;
+		padding-right: ${9 - SVG_PADDING_IN_PX}px;
```

Comment:
The idea is that I wanted to introduce some padding around the svg icon such that the tooltip is triggered from a broader area than just the icon itself. 

However, adding padding to the svg would reduce the width of the link bar. Since the link bar has a padding-right which is 9px by default, when I say {9 - SVG_PADDING_IN_PX} I mean that the extra space taken by the svg padding should be removed from the link bar padding.

In fact, I forgot to add the same property to `.MuiOutlinedInput-input` (which I have now added). This is because the padding used by the svg icon has to be multiplied by 2 (padding right + padding left). Removing the same amount twice from the link bar padding is a way to maintain the same original design of the `Textfield` while making the svg icon tooltip easier to trigger

Reply:
Code refactored